test/unittests/eof_validation_test.cpp

```diff
@@ -58,145 +58,316 @@ TEST(eof_validation, validate_EOF_version)
     EXPECT_EQ(validate_eof("EF00FF"), EOFValidationError::eof_version_unknown);
 
     // valid except version
-    EXPECT_EQ(validate_eof("EF00000 10003 020004 00 600000 AABBCCDD"),
+    EXPECT_EQ(validate_eof("EF0000 010004 0200010003 020004 00 00000000 600000 AABBCCDD"),
         EOFValidationError::eof_version_unknown);
-    EXPECT_EQ(validate_eof("EF00020 10003 020004 00 600000 AABBCCDD"),
+    EXPECT_EQ(validate_eof("EF0002 010004 0200010003 020004 00 00000000 600000 AABBCCDD"),
         EOFValidationError::eof_version_unknown);
-    EXPECT_EQ(validate_eof("EF00FF0 10003 020004 00 600000 AABBCCDD"),
+    EXPECT_EQ(validate_eof("EF00FF 010004 0200010003 020004 00 00000000 600000 AABBCCDD"),
         EOFValidationError::eof_version_unknown);
 }
 
 TEST(eof_validation, valid_EOF1_code_pre_shanghai)
 {
-    EXPECT_EQ(
-        validate_eof("EF0001 010001 00 FE", EVMC_PARIS), EOFValidationError::eof_version_unknown);
+    EXPECT_EQ(validate_eof("EF0001 010004 0200010001 00 00000000 FE", EVMC_PARIS),
+        EOFValidationError::eof_version_unknown);
 }
 
 TEST(eof_validation, minimal_valid_EOF1_code)
 {
-    EXPECT_EQ(validate_eof("EF0001 010001 00 FE"), EOFValidationError::success);
+    EXPECT_EQ(validate_eof("EF0001 010004 0200010001 030000 00 00000000 FE"),
+        EOFValidationError::success);
 }
 
 TEST(eof_validation, minimal_valid_EOF1_code_with_data)
 {
-    EXPECT_EQ(validate_eof("EF0001 010001 020001 00 FE DA"), EOFValidationError::success);
+    EXPECT_EQ(validate_eof("EF0001 010004 0200010001 030001 00 00000000 FE DA"),
+        EOFValidationError::success);
+}
+
+// Types section is mandatory
+// TEST(eof_validation, minimal_valid_EOF1_code_with_type)
```

Comment:
Why is this disaabled?

Reply:
Type section is mandatory. So it would be the same test case as `minimal_valid_EOF1_code `